src/renderer/qssWidget/js/qssOfficeWidget.js

```diff
@@ -0,0 +1,391 @@
+                    "{arguments}.0" // keyboardEvent
+                ]
+            }
+        },
+        events: {
+            onHeightChanged: null
+        }
+    });
+
+
+    /**
+     * Checks the current states and generates the appropriate command
+     * @param {Object} states - simple true/false object with the current states
+     * @param {Object} availableCommands - a simple list of available commands defined in the model
+     * @return {String} - string value of the compiled command
+     */
+    gpii.qssWidget.office.getCommand = function (states, availableCommands) {
+        var stateNames = [],
+            defaultCommand = availableCommands.defaultCommand,
+            resetCommand = availableCommands.resetCommand,
+            allTrue = availableCommands.allTrueCommand,
+            allFalse = availableCommands.allFalseCommand,
+            allStates = 0;
+
+        fluid.each(states, function (state, name) {
+            if (name !== resetCommand) {
+                if (state === true) {
+                    stateNames.push(name);
+                }
+                allStates++;
+            }
+        });
+
+        if (stateNames.length === 0) {
+            // no option is selected
+            return allFalse;
+        } else if (stateNames.length === allStates) {
+            // all of the options are selected
+            return allTrue;
+        } else if (stateNames.length === 1) {
+            // only one of the options is selected
+            // IMPORTANT: returning the name of the option as command
+            return stateNames.pop();
+        } else {
+            // IMPORTANT: we should never got to here
+            // but just in case, returning the default command
+            return defaultCommand;
+        }
+    };
+
+    /**
+     * Pre-loads the data in the office.model.states array
+     * @param {Component} that - The `gpii.qssWidget.office.presenter` instance.
+     * @param {Component} office - The `gpii.qssWidget.office` instance.
+     * @param {String} ribbonState - The initial state of ribbons
+     */
+    gpii.qssWidget.office.loadState = function (that, office) {
+        // pre-fills the states of all available schema keys
+        if (office.model.setting.value === office.model.availableCommands.allTrueCommand) {
+            fluid.each(office.model.setting.schema.keys, function (key) {
+                if (key !== office.model.availableCommands.resetCommand) {
+                    office.model.states[key] = true;
+                }
+            });
+        } else if (office.model.setting.value === office.model.availableCommands.allFalseCommand) {
+            fluid.each(office.model.setting.schema.keys, function (key) {
+                if (key !== office.model.availableCommands.resetCommand) {
+                    office.model.states[key] = false;
+                }
+            });
+        } else {
+            fluid.each(office.model.setting.schema.keys, function (key) {
```

Comment:
The insides of each of these 3 branches are nearly identical, refactor into a shared function

Reply:
Done.